pkg/ir/xds.go

```diff
@@ -0,0 +1,76 @@
+package ir
+
+import (
+	route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+)
+
+// Xds holds the intermediate representation of a Gateway and is
+// used by the xDS Translator to convert it into xDS resources.
+type Xds struct {
+	// Name of the Xds IR.
+	Name string
+	// HTTP listeners exposed by the gateway.
+	HTTP []HTTPListener
+}
+
+// HTTPListener holds the listener configuration.
+type HTTPListener struct {
+	// Name of the HttpListener
+	Name string
+	// Address that the listener should listen on.
+	Address string
+	// Port on which the service can be expected to be accessed by clients.
+	Port uint32
+	// Hostnames (Host/Authority header value) with which the service can be expected to be accessed by clients.
+	// This field is required. Wildcard hosts are supported in the suffix or prefix form.
+	// Refer to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-virtualhost
+	// for more info.
+	Hostnames []string
+	// Tls certificate info. If omitted, the gateway will expose a plain text HTTP server.
+	TLS TLSListenerConfig
+	// Routes associated with HTTP traffic to the service.
+	Routes []HTTPRoute
+}
+
+// HTTPRoute holds the route information associated with the HTTP Route
+type HTTPRoute struct {
+	// Name of the HttpRoute
+	Name string
+	// Matches define the match conditions for this route.
+	Matches []route.HeaderMatcher
```

Comment:
was reducing the number of datatypes to create and mantain ,  can you elaborate on the decoupling point

Reply:
I guess in my mind the whole point of the IR is to decouple what's upstream of it from what's downstream of it -- so the Gateway API translator doesn't need to know about Envoy config, and the xDS translator doesn't need to know about Gateway API config. Using Envoy types here means that the Gateway API translator now has at least _some_ responsibility for directly generating Envoy config, which starts to erode the value of having an IR at all - taken to the extreme, we should just have the Gateway API translator directly generate _all_ Envoy config. 

I can definitely be convinced that for some basic types, it's more expedient to use Envoy's directly in the IR rather than re-creating our own duplicate copies, but I do think it should be more of the exception than the rule. Anyway, just my 2c, others can weigh in as well.

Reply:
I tend to agree with @skriss – using the Envoy type directly forces the IR to update to track Envoy versions, which is probably less than ideal.